### PR TITLE
Disable next button while fetching

### LINKED ISSUE #1: Disable 'next' button while fetching
Remember that a fetch is in progress and disable the button until the fetch is complete.

### PR DESCRIPTION
**Pull request by fedeisas**
Fixes #1 

I'm just learning some React/Redux concepts so I'll really appreciate your input. Thanks!

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -49,20 +49,29 @@ let testPeople = [
...
 export function getRandomPerson(filter = 'all') {
-  if ( process.env.REACT_APP_USE_TEST_DATA === "true") {
-    let data = testPeople.slice().sort(function(a,b) {return Math.random()-Math.random();}).slice(0,4);
-    return new Promise((res, rej) => res(data));
+  if (process.env.REACT_APP_USE_TEST_DATA === 'true') {
+    let data = testPeople.slice().sort(function(a, b) {
+      return Math.random() - Math.random();
+    }).slice(0, 4);
+    return new Promise(async function (res, rej) {
+      await sleep(1000);
```
Comment (jasonaowen) [inline src/api.js]: Why `sleep` when using test data? This doesn't seem to follow the `loading` flow, so I'm not clear on the benefit.

File: src/reducer.js [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
+    case 'HINT':
+      return hint(state);
+    case 'CLEARHINT':
+      return clearHint(state);
+    case 'START_LOADING':
+      return startLoading(state);
+    case 'END_LOADING':
+      return endLoading(state);
+    default:
+      return state;
```
Comment (jasonaowen) [inline src/reducer.js]: There's a lot of refactoring going on, which somewhat obscures the relevant changes. In general, I suggest doing one refactor or change per commit, so maybe string quote changes (`"` -> `'`) in one commit, whitespace changes in another, and then the loading workflow in a third.

File: src/api.js (hunk 1 above)
Comment (fedeisas) [inline src/api.js]: @jasonaowen loading test data is so fast that I couldn't test the desired behaviour otherwise.

**Comment by fedeisas** [comment]
I've removed all the styling changes. I can make another PR with code styling changes?

**Comment by jasonaowen** [comment]
This is great. Thank you so much!

**Pull request merged**